### ISSUE TITLE
When creating a lead field type of boolean, checking

### ISSUE THREAD

**Issue by zapur1** [mautic/mautic#262]
Hi,

When you create a lead field with a Boolean data type it doesn't seem to do any sanity checking on the Default value, meaning that you can put anything there.

It means that the Default value isn't actually applied to a new lead as it is the wrong data type.

What are the default values for a Boolean in the short-term?

**Comment by zapur1** [comment]
It is looking more like the default value isn't being applied until the user first edits the lead. Up until then the value is empty. 

I.e. Anonymous lead created by tracking, no default value
Is edited by a user, default values are applie

**Comment by alanhartless** [comment]
Hi Zapur,

Try using a 1 or 0 as the default value for a boolean field.

I think that you are right that default values are only populating the forms, which was it's original intent but the UI is not making that clear. Marking this as a feature/enhancement to look into getting those two cleared up.

**Comment by zapur1** [comment]
Hi,
I was noticing this when I was grabbing data through the API which might be the reason I'm seeing it differently. When the lead is created from tracking data if I look in the MySQL database the lead has nothing set for the field (I know it is really picky but does make a difference). 

I have tried it with the default value as 0 but am still getting blank fields in the MySQL database until I go into edit the lead and save.

I'm thinking it needs to be a change to how default values are populated when the lead is automatically generated?

I've created a MySQL trigger to fix it up in the mean time
